### PR TITLE
Resource View

### PR DESCRIPTION
**Pull request by mythilytm**
Primary reviewer:

## Description
- This PR modifies the 'ViewResource' component and adds functionality to use the updated data model for resources. 
- Mocks are here: https://marvelapp.com/prototype/2i5hhd8g/screen/90702290
Screenshot 2023-03-17 at 11 47 07 AM

### Checklist
- [x] Corresponding issue has been opened
- [n/a] New tests added
- [n/a] Feature flags added
- [n/a] Strings are localized
- [na] Tested for chat contacts
- [n/a] Tested for call contacts

### Related Issues
Fixes # 1172

### Verification steps
 - Check the first five resources by search
- Compare with the mocks: https://marvelapp.com/prototype/2i5hhd8g/screen/90702290

### CODE REVIEW

File: plugin-hrm-form/src/components/resources/resourceview/ViewResource.tsx [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+                    <ResourceAttribute
+                      description="Details"
+                      content={handleDescriptionInfo(resource.attributes.description)}
+                    />
+                  </ResourceAttributesColumn>
+
+                  {/* SECOND COLUMN */}
+                  <ResourceAttributesColumn>
+                    <ResourceAttributeWithPrivacy
+                      isPrivate={getSingleStringVal(resource.attributes.mainContact, 'isPrivate') === 'false'}
```
Comment (stephenhand) [inline plugin-hrm-form/src/components/resources/resourceview/ViewResource.tsx:183]:
```suggestion
                      isPrivate={getSingleStringVal(resource.attributes.mainContact, 'isPrivate') === 'true'}
```

Did you flip it for testing? everything is showing as private now and I don't think any of the test data is flagged private

**Review by stephenhand — COMMENTED** [review]
Only found one actual incorrect behaviour, but suggested some refactoring

File: plugin-hrm-form/src/components/resources/resourceview/ResourceAttribute.tsx [hunk 2]
```diff
@@ -39,7 +39,7 @@ const isCategory = (c: any): c is Category =>
 
 const ResourceAttribute: React.FC<Props> = ({ description, content }) => {
   const renderContent = () => {
-    if (typeof content === 'string') {
+    if (typeof content === 'string' && description === 'Details') {
```
[the remark this answers is not in the document]
Comment (mythilytm) [reply plugin-hrm-form/src/components/resources/resourceview/ResourceAttribute.tsx]: Thanks! I moved all the logic to `convertResourceData`.

File: plugin-hrm-form/src/components/resources/resourceview/ViewResourceSiteDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+type Props = {
+  attributes: any;
+};
+
+const ViewResourceSiteDetails: React.FC<Props> = ({ attributes }) => {
+  const siteId = Object.keys(attributes.site)[0];
+  const { name, location, email, phone, operations } = attributes.site[siteId];
+
+  const phoneTypes = [
+    { type: 'tollFree', label: 'Toll Free' },
```
[the remark this answers is not in the document]
Comment (mythilytm) [reply plugin-hrm-form/src/components/resources/resourceview/ViewResourceSiteDetails.tsx]: Removed phones as this is not required at the time.

File: plugin-hrm-form/src/components/resources/resourceview/ResourceAttribute.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (mythilytm) [reply plugin-hrm-form/src/components/resources/resourceview/ResourceAttribute.tsx]: Added `isExpandable` flag

File: plugin-hrm-form/src/types/types.ts [hunk 4]
```diff
@@ -308,3 +308,60 @@ export function isTwilioTask(task: CustomITask): task is ITask {
 export const isStandaloneITask = (task): task is StandaloneITask => {
   return task && task.taskSid === standaloneTaskSid;
 };
+
+export type Resource = {
```
Comment (stephenhand) [inline plugin-hrm-form/src/types/types.ts:312]: Maybe rename this to `KhpResource` or `KhpUiResource` so we know this is specifically for displaying KHP resources,

**Review by stephenhand — APPROVED** [review]
Few comments, mostly minor stuff - nice work

**Pull request merged**